src/Abp.Web/Web/AbpModuleAssemblyFindler.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Abp.Collections.Extensions;
+using Abp.Modules;
+using Abp.Reflection;
+
+namespace Abp.Web
+{
+    /// <summary>
+    /// This class is used to get assemblies, which contain at least one AbpModule-derived class.
+    /// </summary>
+    public class AbpModuleAssemblyFindler : IAssemblyFinder
+    {
+        private List<Assembly> _assemblies;
+
+        private readonly object _syncLock = new object();
+
+        /// <summary>
+        /// This return assemblies, which contain at least one AbpModule-derived class
+        /// </summary>
+        /// <returns>List of assemblies</returns>
+        public List<Assembly> GetAllAssemblies()
+        {
+            if (_assemblies == null)
+            {
+                lock (_syncLock)
+                {
+                    if (_assemblies == null)
+                    {
+                        _assemblies = GetAbpModuleAssembliesRecursive();
+                    }
+                }
+            }
+
+            return _assemblies;
+        }
+
+        /// <summary>
+        /// Retrieves all of the AbpModules and their dependencies
+        /// </summary>
+        /// <returns></returns>
+        private List<Assembly> GetAbpModuleAssembliesRecursive()
+        {
+            var currentAssembly = Assembly.GetCallingAssembly();
```

Comment:
In the [document](https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getcallingassembly(v=vs.110).aspx), it says "Returns the Assembly of the method that invoked the currently executing method"

So, doesn't that returns the Abp assembly, since caller method is GetAllAssemblies which is in Abp package?